### PR TITLE
bpo-30627: Fix error message when keyword arguments are used

### PR DESCRIPTION
**Pull request by SylvainDe**

### CODE REVIEW

File: Modules/_struct.c [hunk 1]
```diff
@@ -1832,9 +1832,6 @@ s_pack(PyObject *self, PyObject **args, Py_ssize_t nargs, PyObject *kwnames)
             "pack expected %zd items for packing (got %zd)", soself->s_len, nargs);
         return NULL;
     }
-    if (!_PyArg_NoStackKeywords("pack", kwnames)) {
```
Comment (serhiy-storchaka) [inline Modules/_struct.c:1835]: Now `Struct.pack()` just ignores keyword arguments.

Comment (SylvainDe) [reply Modules/_struct.c:1835]: Indeed, I did not know this could be called via a different path. Thanks for spotting this.

Comment (SylvainDe) [reply Modules/_struct.c:1835]: I've updated the PR, thanks for catching this!

File: Modules/_struct.c [hunk 2]
```diff
@@ -2131,6 +2131,10 @@ pack(PyObject *self, PyObject **args, Py_ssize_t nargs, PyObject *kwnames)
     PyObject *s_object = NULL;
     PyObject *format, *result;
 
+    if (!_PyArg_NoStackKeywords("pack", kwnames)) {
```
Comment (serhiy-storchaka) [inline Modules/_struct.c:2134]: Is this needed?

Comment (SylvainDe) [reply Modules/_struct.c:2134]: It is needed if we want the `pack` (and `pack_into`) function to have the same behavior as other function, ie checking for (unsupported) keyword arguments is the first check we perform. If we remove it, the tests `test_varargs8_kw` and `test_varargs9_kw` fail with error "missing format argument" as the check for keyword happened afterward (in the "s_"-prefixed functions).
I'm happy to get rid of the check and remove/update the test if you think it is better.

Thanks for your feedback!

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**